dlib/simd/simd8f_vec.h

```diff
@@ -0,0 +1,251 @@
+// Copyright (C) 2013  Davis E. King (davis@dlib.net)
+// License: Boost Software License   See LICENSE.txt for the full license.
+#ifndef DLIB_sIMD8F_Hh_
```

Comment:
Ha, just saw your next change. :)   All in one file is fine too.  I think it's easy to read now.  There is the worry that it might become horrendous at some point in the future, but now it's good.  And it has the benefit at least that when new architectures are added it's difficult to accidentally forget to add some part of the simd*f API.